### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by p-sam**
PR including the changes I made to build on Linux, for the libretro core (Switch target) and the standalone version using CMake

### CODE REVIEW

File: src/views/view_manager.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ void ui::ViewManager::text(const std::string& text, int32_t x, int32_t y)
   for (size_t i = 0; i < text.length(); ++i)
   {
     SDL_Rect src = { 8 * (text[i] % GLYPHS_PER_ROW), 8 * (text[i] / GLYPHS_PER_ROW), 4, 6 };
-    SDL_Rect dest = { x + 4 * i * scale, y, 4 * scale, 6 * scale };
+    SDL_Rect dest = { static_cast<int>(x + 4 * i * scale), y, static_cast<int>(4 * scale), static_cast<int>(6 * scale) };;
```
Comment (Jakz) [inline src/views/view_manager.cpp]: add a `inline static SDL_Rect SDL_MakeRect(int x, int y, int w, int h) { return { x, y, w, h }; }` to SDL_Helper.h and use the function instead that casting around

File: CMakeLists.txt [hunk 2]
```diff
@@ -22,7 +26,7 @@ add_compile_options(-Wno-unused-parameter -Wno-missing-field-initializers
 add_compile_options(-g -O2 -W -Wall -Wextra)
 add_compile_options(
   $<$<COMPILE_LANGUAGE:CXX>:-Wno-reorder>
-  $<$<COMPILE_LANGUAGE:CXX>:-std=c++11>
+  $<$<COMPILE_LANGUAGE:CXX>:-std=c++14>
```
Comment (Jakz) [inline CMakeLists.txt]: C++14 can't be used because OpenDingux only has C++11 support.

Comment (p-sam) [reply CMakeLists.txt]: Conditionned to the `APPLE` cmake var now, as right now only MacOS doesn't have it on C++11 too

**Comment by p-sam** [comment]
There's a few other SDL_Rect declarations that could use being done using the inline SDL_MakeRect func, since it's there now

**Pull request merged**